File: homework10.js
function onErr(err) {
  console.log('>>>' + err + '<<< is not a valid input');
  return 1;
}
function numToWord(intVal) {
  var place = intVal.length;
  var totalPlace = place;
  var placeOne = '',
    placeTwo = '',
    placeThree = '',
    placeFour = '';
  var ones = false,
    tens = false,
    teen = false,
    hundreds = false,
    thousands = false;

  var teenHolder = '';
  var placeHolder = intVal.split("");
  for (var k = 0; k < placeHolder.length; k++) {

    if (place == 4) { // Algorithm for changing the numbers to words
      if (placeHolder[k] == 9) {
        placeOne = 'nine thousand';
        thousands = true;
      } else if (placeHolder[k] == 8) {
        thousands = true;
        placeOne = 'eight thousand';
        thousands = true;
      } else if (placeHolder[k] == 7) {
        placeOne = 'seven thousand';
        thousands = true;
      } else if (placeHolder[k] == 6) {
        placeOne = 'six thousand';
        thousands = true;
      } else if (placeHolder[k] == 5) {
        placeOne = 'five thousand';
        thousands = true;
      } else if (placeHolder[k] == 4) {
        placeOne = 'four thousand';
        thousands = true;
      } else if (placeHolder[k] == 3) {
        placeOne = 'three thousand';
        thousands = true;
      } else if (placeHolder[k] == 2) {
        placeOne = 'two thousand';
        thousands = true;
      } else if (placeHolder[k] == 1) {
        placeOne = 'one thousand';
        thousands = true;
      }

    } else if (place == 3) {
      if (placeHolder[k] == 9) {
        placeTwo = 'nine hundred';
        hundreds = true;
      } else if (placeHolder[k] == 8) {
        placeTwo = 'eight hundred';
        hundreds = true;
      } else if (placeHolder[k] == 7) {
        placeTwo = 'seven hundred';
        hundreds = true;
      } else if (placeHolder[k] == 6) {
        placeTwo = 'six hundred';
        hundreds = true;
      } else if (placeHolder[k] == 5) {
        placeTwo = 'five hundred';
        hundreds = true;
      } else if (placeHolder[k] == 4) {
        placeTwo = 'four hundred';
        hundreds = true;
      } else if (placeHolder[k] == 3) {
        placeTwo = 'three hundred';
        hundreds = true;
      } else if (placeHolder[k] == 2) {
        placeTwo = 'two hundred';
        hundreds = true;
      } else if (placeHolder[k] == 1) {
        placeTwo = 'one hundred';
        hundreds = true;
      } else if (placeHolder[k] == 0) {

      }

    } else if (place == 2) {
      if (placeHolder[k] == 9) {
        placeThree = 'ninety';
        tens = true;
      } else if (placeHolder[k] == 8) {
        placeThree = 'eighty';
        tens = true;
      } else if (placeHolder[k] == 7) {
        placeThree = 'seventy';
        tens = true;
      } else if (placeHolder[k] == 6) {
        placeThree = 'sixty';
        tens = true;
      } else if (placeHolder[k] == 5) {
        placeThree = 'fifty';
        tens = true;
      } else if (placeHolder[k] == 4) {
        placeThree = 'fourty';
        tens = true;
      } else if (placeHolder[k] == 3) {
        placeThree = 'thirty';
        tens = true;
      } else if (placeHolder[k] == 2) {
        placeThree = 'twenty';
        tens = true;
      } else if (placeHolder[k] == 1) {
        if (placeHolder[k + 1] == 9) {
          teenHolder = 'nineteen';
                  teen = true;
        } else if (placeHolder[k + 1] == 8) {
          teenHolder = 'eighteen';
                  teen = true;
        } else if (placeHolder[k + 1] == 7) {
          teenHolder = 'seventeen';
                  teen = true;
        } else if (placeHolder[k + 1] == 6) {
          teenHolder = 'sixteen';
                  teen = true;
        } else if (placeHolder[k + 1] == 5) {
          teenHolder = 'fifteen';
                  teen = true;
        } else if (placeHolder[k + 1] == 4) {
          teenHolder = 'fourteen';
                  teen = true;
        } else if (placeHolder[k + 1] == 3) {
          teenHolder = 'thirteen';
                  teen = true;
        } else if (placeHolder[k + 1] == 2) {
          teenHolder = 'twelve';
                  teen = true;
        } else if (placeHolder[k + 1] == 1) {
          teenHolder = 'eleven';
                  teen = true;
        } else if (placeHolder[k + 1] == 0) {
          teenHolder = 'ten'
                  teen = true;
        }

      }
    } else if (place == 1) {
      if (placeHolder[k] == 9) {
        placeFour = 'nine';
        ones = true;
      } else if (placeHolder[k] == 8) {
        placeFour = 'eight';
        ones = true;
      } else if (placeHolder[k] == 7) {
        placeFour = 'seven';
        ones = true;
      } else if (placeHolder[k] == 6) {
        placeFour = 'six';
        ones = true;
      } else if (placeHolder[k] == 5) {
        placeFour = 'five';
        ones = true;
      } else if (placeHolder[k] == 4) {
        placeFour = 'four';
        ones = true;
      } else if (placeHolder[k] == 3) {
        placeFour = 'three';
        ones = true;
      } else if (placeHolder[k] == 2) {
        placeFour = 'two';
        ones = true;
      } else if (placeHolder[k] == 1) {
        placeFour = 'one';
        ones = true;
      }

    }

    place--;
  }
  console.log('\n' + intVal + ' is turned into: ');
  if (totalPlace == 4) { // Yes Thousands
    if (hundreds == true) { // Yes Thousands Yes Hundreds
      if (teen == true) { // Yes Thousands Yes Hundreds Yes Teens
        console.log(placeOne + ' ' + placeTwo + ' ' + teenHolder);
      } else if (tens == true) { // Yes Thousands Yes Hundreds Yes Tens
        if (ones == true) { // Yes Thousands Yes Hundreds Yes Tens Yes Ones
          console.log(placeOne + ' ' + placeTwo + ' ' + placeThree + ' ' + placeFour);
        } else { // Yes Thousands Yes Hundreds Yes Tens No Ones
          console.log(placeOne + ' ' + placeTwo + ' and ' + placeThree);
        }
      } else if (tens == false && teen == false) { // Yes Thousands Yes Hundreds, Nothing else
        console.log(placeOne + ' ' + placeTwo);
      } else { // Error
        onErr(intVal);
      }
    } else { // Yes Thousands No Hundreds
      if (teen == true) { // Yes Thousands No Hundreds Yes Teens
        console.log(placeOne + ' and ' + teenHolder);
      } else if (tens == true) { // Yes Thousands No Hundreds Yes Tens
        if (ones == true) { // Yes Thousands No Hundreds Yes Tens Yes Ones
          console.log(placeOne + ' ' + placeThree + ' ' + placeFour);
        } else { // Yes Thousands No Hundreds Yes Tens No Ones
          console.log(placeOne + ' and ' + placeThree);
        }
      } else if (tens == false && ones == true) { // Yes Thousands No Hundreds No Tens Yes Ones
        console.log(placeOne + ' and ' + placeFour);
      } else { // Error
        onErr(intVal);
      }
    }
  } else if (totalPlace == 3) { // No Thousands Yes Hundreds
    if (teen == true) {
      console.log(placeTwo + ' ' + teenHolder);
    } else if (tens == true) {
      if (ones == true) {
        console.log(placeTwo + ' ' + placeThree + ' ' + placeFour);
      } else {
        console.log(placeTwo + ' ' + placeThree)
      }
    } else {
      console.log(placeTwo);
    }
  } else if (totalPlace == 2) { // No Thousands No Hundreds Yes Tens / Teens
    if (teen == true) { // Yes Teens
      console.log(teenHolder);
    } else { // Yes Tens
      if (ones == true) { // Yes Tens Yes Ones
        console.log(placeThree + ' ' + placeFour);
      } else { // Yes Tens No Ones
        console.log(placeThree);
      }
    }
  } else if (totalPlace == 1) { // No Thousands No Hundreds No Tens / Teens Yes Ones
    console.log(placeFour);
  } else {
    onErr(intVal);
  }

} // End of numToWord();

function main() {
  var fileName = 'homework10.data'; // Process homework10.data into a string
  
  // Download [generic].data
  // Split [generic].data
  var __dirname = "C:\\Users\\ben\\src\\dadschool2015";

  var fs = require('fs');
  var path = require('path');

  var filePath = path.join(__dirname, fileName);

  fs.readFile(filePath, { encoding: 'utf-8' }, function (err, data) {// Start of file download code
    if (!err) {
      console.log("Now downloading data from " + filePath);
      console.log('>>>\n' + data + "\n<<<");
    }
    else {
      console.log(err);
    }
    var dataNumber = data.split(/\s+/);
    // Loop through [generic].data
    for (var i = 0; i < dataNumber.length; i++) {
      var currentNum = dataNumber[i];
      numToWord(currentNum);
    }
  }); // End of file download code
  
}
main(); // Run program

// The program has finished!

File: homework10.cpp
/*******************************************************************************
********************************************************************************
**
**  Nathan A. Zobrist
**  Assignment #10
**  CSC 135, Section 1
**
********************************************************************************
**
**  This program is designed to read data from a file named progAss10.data.
**    The data file should consist only of intigers between 1 and 9999.  The
**    program will convert the number to text and print it to the screen.
**    the program will process data until the EOF is reached.
**
**  input: whole number between 1 and 9999
**  output: the whole number converted to text formatted as per assignment
**
********************************************************************************
*******************************************************************************/


#include <iostream>
#include <string>       // 2 functions of type string
#include <fstream>      // to read the data file, "progAss10.data"

string convertToText (int num);
string digitToText (int num);

using namespace std;

int main ()
{
    ifstream infile;
    
    int num = 0;        // in main (), num will be the number read from the
                        // data file which is to be converted to text

    infile.open("progAss10.data");

    infile >> num;

    while(infile)
    {
        cout << endl << num << " = " << convertToText(num);
        cout << endl;
        infile >> num;
    }
    
    cout << endl;
    
    return 0;

}

string convertToText (int num)
/*******************************************************************************
**
**  The function convertToText is designed to convert any intiger between
**    1 and 9999 into text.
**
**  input: one whole number between 1 and 9999
**  output: convertToText() will return a string containing the text equivalent
**    of the number it was passed.
**
*******************************************************************************/
{
    int thousands, hundreds, tens;  // used to break the number down into 
                                    // smaller parts
    string final;      // final will be returned as the result of the function
    
    if (num >= 1000)
    {
        thousands = num / 1000;
        final += digitToText(thousands) + " thousand";
        num %= 1000;
    }

    if (num >= 100)
    {
        hundreds = num / 100;
        final += digitToText(hundreds) + " hundred";
        num %= 100;
    }

    if (num >= 20)
    {
        tens = num / 10;
        final += digitToText(tens * 10);
        num %= 10;
    }

    final += digitToText(num);
    
    return final;
}

string digitToText(int num)
/*******************************************************************************
**
**  The function digitToText is designed to convert the numbers 1 - 19 and 
**    multiples of 10 up to 90 into text.
**
**  input: one integer (1-19 or 20-90, multiples of 10)
**  output: a string of text equivalent to the number which was passed
**
*******************************************************************************/
{
    switch(num)
    {
        case 1: return " one";
        case 2: return " two";
        case 3: return " three";
        case 4: return " four";
        case 5: return " five";
        case 6: return " six";
        case 7: return " seven";
        case 8: return " eight";
        case 9: return " nine";
        case 10: return " ten";
        case 11: return " eleven";
        case 12: return " twelve";
        case 13: return " thirteen";
        case 14: return " fourteen";
        case 15: return " fifteen";
        case 16: return " sixteen";
        case 17: return " seventeen";
        case 18: return " eighteen";
        case 19: return " nineteen";
        case 20: return " twenty";
        case 30: return " thirty";
        case 40: return " forty";
        case 50: return " fifty";
        case 60: return " sixty";
        case 70: return " seventy";
        case 80: return " eighty";
        case 90: return " ninety";
    }
}
